File: in_Java/0290 Word Pattern.java
class Solution {
    public boolean wordPattern(String pattern, String str) {
        char[] ch = pattern.toCharArray();
        String[] st = str.split(" ");
        if(ch.length != st.length) return false;
        HashMap<Character,String> hm = new HashMap<>();
        for(int i = 0; i < ch.length; i++){
            char a = ch[i];
            String b = st[i];
            if(!hm.containsKey(a)){
                if(hm.containsValue(b)) return false;
                else hm.put(a,b);
            }else{
                if(!hm.get(a).equals(b)) return false;
            }
        }
        return true;
    }
}

File: in_Python/0290 Word Pattern.py
class Solution(object):
    def wordPattern(self, pattern, s):
        """
        :type pattern: str
        :type s: str
        :rtype: bool
        """
        if pattern == None and s == None:
            return True
        if pattern == None or s == None:
            return False
        p = s.split(' ')
        if len(pattern) != len(p):
            return False
        dict = {}
        for i in range(len(p)):
            k = pattern[i]
            v = p[i]
            if k in dict and dict[k] != v:
                return False
            for key in dict:
                if dict[key] == v and key != k:
                    return False
            if k not in dict:
                dict[k] = v
        return True
            
                
        